docs/src/conceptual/solution-selection-catalogs.rst

```diff
@@ -0,0 +1,181 @@
+.. meta::
+  :description: Tensile is a tool for creating a benchmark-driven backend library for GEMM
+  :keywords: Tensile, GEMM, Tensor, Tensile API documentation, Tensile library creation
+
+.. _solution-catalogs:
+
+***************************
+Solution selection catalogs
+***************************
+
+After kernels are compiled and linked into code objects libraries (.co files), we still have the problem of how these kernels are executed at runtime. The naive approach
+would be to search through all of the code object libraries until an appropriate kernel is found. A more sophisticated approach is to use a heirarchical structure
+that allows client code to search for kernels using predicates such as hardware, problem size, and transpose, among others. 
+This is the role of the **solution selection catalog** [1]_---a serialized file that uses a heirarchical schema to organize kernel metadata for efficient lookup at runtime.
+
+.. note::
+    Throughout this document we will refer to catalog files with the .yaml extension. In practice
+    solution selection catalogs are usually serialized with `MessagePack <https://msgpack.org/>`_, which uses the .dat extension.
+
+Catalog hierarchy
+=================
+
+.. figure:: ../../assets/msl.svg
+    :alt: Master Solution Library hierarchy
+    :align: center
+
+    Solution selection catalog heirarchy for gfx900 and gfx90a
+
+**Level 1: Hardware**
+
+At runtime, only kernels compatible with the host machine's installed hardware can execute. As such, the top level of the hierarchy involves hardware comparisions using GFX architecture.
+
+**Level 2: Operation**
+
+This layer is a mapping from a GEMM transpose setting, defined using 
+Einstein tensor notation (e.g. *Contraction_l_Alik_Bjlk_Cijk_Dijk*), to a list of problem properties (Level 3).
+
+**Level 3: Problem**
+
+This layer matches against specific problem properties such as input and output types, and features like high precision accumulation and stochastic rounding.
+
+**Level 4: Exact solution**
+
+Finally, exact solutions contain fine-grained details about each solution that can be used during solution selection to locate the best kernel and to assert that the requested problem predicates are satisfied. At this level, there will typically exist a small pool of acceptable kernels for any requested GEMM. Each kernel will have an index and a performance ranking. During solution selection, this highest ranked kernel from this pool will be selected.
```

Comment:
```suggestion
Finally, exact solutions contain fine-grained details about each solution that can be used during solution selection to locate the best kernel and to assert that the requested problem predicates are satisfied. Each kernel will have an index and a performance ranking. During solution selection, the highest ranked kernel from this pool will be selected.
```
> During solution selection, the highest ranked kernel from this pool will be selected.

Is this only if there is a math in the pool?

> Each kernel will have an index and a performance ranking

This sounds like it is only true for equality matches.
>